### PR TITLE
feat: ensure pair aex9 compliance 

### LINKED ISSUE #15: Ensure Pair AEX9 compatibility
based on [https://github.com/aeternity/dex-contracts-v2/pull/10#discussion_r765966522](https://github.com/aeternity/dex-contracts-v2/pull/10#discussion_r765966522)

### LINKED ISSUE #30: refactor calling entrypoints in tests
- replace old style (exe) of calling entrypoints with new one from the fixtures
- remove old style from fixtures

### PR DESCRIPTION
**Pull request by bogdan-manole**
Closes #15 
Closes #30

### CODE REVIEW

**Comment by kenodressel** [comment]
Can we align this with #32? There seem to be a lot of things here that are already overwritten there....

File: contracts/AedexV2Pair.aes [hunk 1]
```diff
@@ -68,76 +71,55 @@ contract AedexV2Pair =
...
+            , reserve1               = 0
+            , block_timestamp_last   = 0
             , price0_cumulative_last = 0
             , price1_cumulative_last = 0
-            , k_last = 0
-            , debug_time = debug_time
-            , min_liquidity = min_liquidity
+            , k_last                 = 0
+            , debug_time             = debug_time
+            , min_liquidity          = min_liquidity
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply contracts/AedexV2Pair.aes:88]: found it also in our implementation on AEX9 implementation [https://github.com/mradkov/aeternity-fungible-token/blob/dafb9b357636be730150d5916dde0b09065b1c4d/contracts/fungible-token-full.aes#L80](https://github.com/mradkov/aeternity-fungible-token/blob/dafb9b357636be730150d5916dde0b09065b1c4d/contracts/fungible-token-full.aes#L80)

[the remark this answers is not in the document]
Comment (bogdan-manole) [reply contracts/AedexV2Pair.aes:88]: you mean the spaces or the comma?

[the remark this answers is not in the document]
Comment (kenodressel) [reply contracts/AedexV2Pair.aes:88]: Both actually :)

[the remark this answers is not in the document]
Comment (thepiwo) [reply contracts/AedexV2Pair.aes:88]: @kenodressel its common in functional programming, I'm fine for readability of some sections or if it is overall followed

File: contracts/AedexV2Pair.aes [hunk 2]
```diff
@@ -68,76 +71,55 @@ contract AedexV2Pair =
...
-        require(Option.is_some(state.debug_time),"AedexV2Pair: NOT_DEBUG_MODE")
-        put( state{ debug_time= Some(time)  } )
-
-    // update reserves and, on the first call per block, price accumulators
-    stateful function update(balance0: int, balance1: int, reserve0: int,reserve1: int  ) =
-        require( balance0 >= 0 && balance1 >= 0 && reserve0 >= 0 && reserve1 >= 0,"AedexV2Pair: LP" )
+    //------------------------------------------------------------------------------
+    // VIEWERS
+    //------------------------------------------------------------------------------
+    //TODO: we might want as meta_info to reflect token1&token0 meta_info
```
Comment (thepiwo) [inline contracts/AedexV2Pair.aes:94]: this would actually be very good, so it can be distinguished in a gui

File: contracts/AedexV2Pair.aes [hunk 3]
```diff
@@ -68,76 +71,55 @@ contract AedexV2Pair =
...
+    //find a suitable combination
+    entrypoint meta_info(): IAEX9Minimal.meta_info = {
+          name = "Aedex v2",
+          symbol = "AEDX-V2",
+          decimals = 18
+        }
 
-        let block_timestamp = timestamp()
+    //TODO: even if we have mint and burn, the entrypoints signature and business
```
Comment (thepiwo) [inline contracts/AedexV2Pair.aes:102]: I would say to officially not support mintable and burnable then as standardized, so a third party application won't be confused

**Review by thepiwo — APPROVED** [review]

**Review by kenodressel — APPROVED** [review]

**Pull request merged**